### PR TITLE
dns/proxy view

### PR DESCRIPTION
**Pull request by aaronblohowiak**
Creates a new view type called `dns`, which supports a rank-and-file layout and annotations. We should talk about how to document these view-specific metadata and/or if we should life annotations into connectionView. If we do the latter, we may want to parameterize the line thickness.

### CODE REVIEW

**Comment by aaronblohowiak** [comment]
es-lint clean, btw

File: src/dns/dnsConnectionView.js [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+import GlobalStyles from '../globalStyles';
+
+const Console = console;
+
+function distance (a, b) {
+  return Math.sqrt(Math.pow(a.x - b.x, 2) + Math.pow(a.y - b.y, 2));
+}
+
+// Given three points, A B C, find the center of a circle that goes through all three.
+function CalculateCircleCenter (A, B, C) {
```
Comment (jrsquared) [inline src/dns/dnsConnectionView.js]: Can exceed call stack limit if one of the slopes is always 0.
Ex: `CalculateCircleCenter({x:1, y:1}, {x:2, y:1}, {x:3, y:1})`

Should have a base return, or only try three times.

File: src/dns/dnsConnectionView.js [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+
+  cleanup () {
+    super.cleanup();
+    _.each([
+      this.annotationMaterial,
+      this.annotationTexture,
+      this.annotationGeometry,
+      this.annotationMesh
+    ], x => { try { x.dispose(); } catch (e) { Console.log(e); } });
+    // TODO: cleanup annotations.
```
Comment (jrsquared) [inline src/dns/dnsConnectionView.js]: Looks like you already are cleaning up annotations?

**Comment by jrsquared** [comment]
Looks great! Just a couple nits commented.

**Pull request merged**